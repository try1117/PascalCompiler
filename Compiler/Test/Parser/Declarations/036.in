program test;
var
  a, b: integer;
  d: double;
  c: char;
  s: string;

begin
  write(integer(c));
  write((integer('0') + integer('5') + integer(c)) div (a + b));
  write(char(230 - 5 * 25 - 7 * a) + string(c));
end.
