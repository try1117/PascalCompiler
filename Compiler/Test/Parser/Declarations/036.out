test : function()
   resultType : Nil

test declarations:
   a : Integer

   b : Integer

   d : Double

   c : Char

   s : String

|-- Statements
|            |-- Write
|            |       |-- Integer
|            |       |         |-- c
|            |-- Write
|            |       |-- div
|            |       |     |-- +
|            |       |     |   |-- 101
|            |       |     |   --- Integer
|            |       |     |             |-- c
|            |       |     --- +
|            |       |         |-- a
|            |       |         --- b
|            --- Write
|                    |-- +
|                    |   |-- String
|                    |   |        |-- Char
|                    |   |        |      |-- -
|                    |   |        |      |   |-- 105
|                    |   |        |      |   --- *
|                    |   |        |      |       |-- 7
|                    |   |        |      |       --- a
|                    |   --- String
|                    |            |-- c

